src/foam/dao/history/HistoryRecordService.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * @license
+ * Copyright 2023 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.INTERFACE({
+  package: 'foam.dao.history',
+  name: 'HistoryRecordService',
+
+  skeleton: true,
+
+  methods: [
+    {
+      name: 'getRecord',
+      type: 'HistoryRecord',
+      documentation: `
+        Returns the latest record that has 'propertyName' property update
+      `,
+      async: true,
+      args: [
+        { name: 'x',              type: 'Context' },
+        { name: 'dao',            type: 'foam.dao.DAO' },
```

Comment:
Would recommend this be String, as the View will be configured with a daoKey.

Reply:
If we pass dao directly, we can pass filtered dao. For example, if we want to get records for certain user, we can filter the dao with that user's id and that pass that dao to the method. With string, we cannot do that. One way I can think of achieving this with daoKey is to add extra id parameter.

Reply:
How do you envision doing this from a View implementation? 
view: { 
  class: '...HistoryRecordView',
  dao: x.get() where...     
}
And the view should know the property it is for. 
@kgrgreer what can we support at the dao point above?

Reply:
DAO's aren't serializable, so you aren't going to be able to pass a DAO from the client to the server. If you need to ability to supply an mlang, then you would need two properties: daoKey and predicate.

Reply:
replaced dao with daoKey for all the methods and added new methods getRecordById and getRecordsById to support filtering by id.

Reply:
re: by id - add a predicate instead please.
when the predicate is null, we'll use dao.getOf().getAxiomByName("id") to query by id property.